roles/openshift_master/tasks/restart.yml

```diff
@@ -3,20 +3,17 @@
   service:
     name: "{{ openshift_service_type }}-master-api"
     state: restarted
-  when: openshift_master_ha | bool
 - name: Wait for master API to come back online
   wait_for:
     host: "{{ openshift.common.hostname }}"
     state: started
     delay: 10
     port: "{{ openshift.master.api_port }}"
     timeout: 600
-  when: openshift_master_ha | bool
-- name: Restart master controllers
-  service:
-    name: "{{ openshift_service_type }}-master-controllers"
-    state: restarted
-  # Ignore errrors since it is possible that type != simple for
-  # pre-3.1.1 installations.
-  ignore_errors: true
-  when: openshift_master_ha | bool
+# We retry the controllers because the API may not be 100% initialized yet.
+- name: restart master controllers
+  command: "systemctl restart {{ openshift_service_type }}-master-controllers"
```

Comment:
Any reason not to use systemd module?

Reply:
This is copy and pasted to match what we have in the handler.  I think we're using command for the retry ability, I don't recall specifically why but I think there might be an issue with retrying the service module.